src/Symfony/Component/Process/Tests/ProcessTest.php

```diff
@@ -133,6 +133,28 @@ public function testStopWithTimeoutIsActuallyWorking()
         $this->assertLessThan(15, microtime(true) - $start);
     }
 
+    public function testWaitUntilSpecificOutput()
+    {
+        $p = $this->getProcess(array(self::$phpBin, __DIR__.'/KillableProcessWithOutput.php'));
+        $p->start();
+
+        $start = microtime(true);
+
+        $completeOutput = '';
+        $p->waitUntil(function ($type, $output) use (&$completeOutput) {
+            $completeOutput .= $output;
+            if (false !== strpos($output, 'One more')) {
+                return true;
+            }
+
+            return false;
+        });
+        $p->stop();
+
+        $this->assertEquals("First iteration output\nSecond iteration output\nOne more iteration output\n", $completeOutput);
+        $this->assertLessThan(15, microtime(true) - $start);
```

Comment:
`15` is too much, the process do 3 iterations in `KillableProcessWithOutput.php` so it should take less than 2 seconds